File: backend/Irrtirer/Irrtirer.Generator/Helpers/TrigonometryHelper.cs
using System;
using System.Numerics;

namespace Irrtirer.Generator.Helpers
{
    internal class TrigonometryHelper
    {
        const float Deg2Rad = MathF.PI / 180;

        public static Vector2 RotatePointAroundPoint(Vector2 point, Vector2 center, float angle)
        {
            angle *= Deg2Rad;
            float x = MathF.Cos(angle) * (point.X - center.X) - MathF.Sin(angle) * (point.Y - center.Y) + center.X;
            float y = MathF.Sin(angle) * (point.X - center.X) + MathF.Cos(angle) * (point.Y - center.Y) + center.Y;

            return new Vector2(x, y);
        }
    }
}

File: frontend/projects/irrtirer/src/app/core/helpers/polygon/trigonometry-helper.ts
import { Vector } from '../../models/math/vector.model';

const deg2Rad: number = Math.PI / 180;

export function rotatePointAroundPoint(point: Vector, center: Vector, angle: number): Vector {
    angle *= deg2Rad;
    const x: number = Math.cos(angle) * (point.x - center.x) - Math.sin(angle) * (point.y - center.y) + center.x;
    const y: number = Math.sin(angle) * (point.x - center.x) + Math.cos(angle) * (point.y - center.y) + center.y;

    return new Vector(x, y);
}

export function transformPolygon(polygon: Vector[], position: Vector, angle: number = 0, scale: number = 1): Vector[] {
    return polygon
        .map((vertex: Vector) => rotatePointAroundPoint(vertex, Vector.zero, angle)
            .multiply(scale)
            .add(position)
        );
}
